src/libappimage/type1.c

```diff
@@ -110,13 +110,43 @@ void type1_extract_file(appimage_handler* handler, void* data, const char* targe
     close(f);
 }
 
+bool type1_read_file_into_buf(struct appimage_handler* handler, void* data, char** buffer, unsigned long* buf_size) {
+    (void) data;
+
+    struct archive* a = handler->cache;
+
+    struct archive_entry* entry = data;
+
+    int64_t file_size = archive_entry_size(entry);
+
+    char* new_buffer = (char*) malloc(sizeof(char) * file_size);
+
+    if (new_buffer == NULL) {
+#ifdef STANDALONE
+        fprintf(stderr, "failed to allocate enough memory for buffer (required: %ul bytes)\n", file_size);
+#endif
+        return false;
+    }
+
+    if (archive_read_data(a, new_buffer, (size_t) file_size) < 0) {
+#ifdef STANDALONE
+        fprintf(stderr, "failed to read data into buffer: %s\n", archive_error_string(a));
+#endif
+        return false;
```

Comment:
Don't you leak the buffer here, given you don't pass it outside for manual `free()` but you also don't dispose of it in here?

Reply:
Good catch, it must be `free()`d of course.